app/src/main/java/com/x8bit/bitwarden/data/autofill/accessibility/model/KnownUsernameField.kt

```diff
@@ -0,0 +1,71 @@
+package com.x8bit.bitwarden.data.autofill.accessibility.model
+
+/**
+ * Represents the known username fields for a given [uriAuthority].
+ */
+data class KnownUsernameField(
+    val uriAuthority: String,
+    val accessOptions: List<AccessOptions>,
+) {
+    constructor(
+        uriAuthority: String,
+        accessOption: AccessOptions,
+    ) : this(uriAuthority = uriAuthority, accessOptions = listOf(accessOption))
+}
+
+/**
+ * Represents the view IDs for a given uri path.
+ */
+data class AccessOptions(
+    val uriPathWanted: String,
+    val matchingStrategy: MatchingStrategy = MatchingStrategy.ENDS_WITH_CASE_SENSITIVE,
+    val usernameViewIds: List<String>,
+) {
+    constructor(
+        uriPathWanted: String,
+        matchingStrategy: MatchingStrategy = MatchingStrategy.ENDS_WITH_CASE_SENSITIVE,
+        usernameViewId: String,
+    ) : this(
+        uriPathWanted = uriPathWanted,
+        matchingStrategy = matchingStrategy,
+        usernameViewIds = listOf(usernameViewId),
+    )
+
+    /**
+     * Indicates the matching strategy needed for the particular [AccessOptions].
+     */
+    enum class MatchingStrategy(
+        val matches: (uriPath: String, uriPathWanted: String) -> Boolean,
```

Comment:
To make clear that one is a URI path and the other is literally just any value to match on using different sets of rules, can we rename `uriPathWanted` to `matchValue`? Since it can literally be an string you try to match on?